nucypher/policy/conditions/address.py

```diff
@@ -0,0 +1,147 @@
+import json
+from typing import Dict, List, Optional, Tuple, Union
+
+from eth_utils import to_checksum_address
+from marshmallow import fields, validate
+
+from nucypher.policy.conditions.base import AccessControlCondition
+from nucypher.policy.conditions.context import (
+    USER_ADDRESS_CONTEXT,
+    resolve_any_context_variables,
+)
+from nucypher.policy.conditions.exceptions import (
+    InvalidCondition,
+    InvalidConditionContext,
+)
+from nucypher.policy.conditions.lingo import (
+    ConditionType,
+)
+
+
+class AddressAllowlistCondition(AccessControlCondition):
+    """
+    A condition that checks if a user's wallet address is in a list of allowed addresses.
+    The user must provide a signed message to prove ownership of the wallet.
+    """
```

Comment:
WDYT? to avoid using the word wallet as @cygnusv suggested.

```suggestion
    """
    A condition that checks if a user's EOA address is in a list of allowed addresses.
    The user must provide a signed message to prove ownership of the EOA.
    """
```

Reply:
I guess here as we use `wallet "address"` it is descriptive enough, especially that it is for both EOA and Smart Contract Accounts as the authentication supports smart contract signature as well. Or what do you think?

Reply:
> I guess here as we use `wallet "address"` it is descriptive enough, especially that it is for both EOA and Smart Contract Accounts as the authentication supports smart contract signature as well. Or what do you think?

it sounds good to me!